### DISCUSSION TITLE
Lobechat usable with cli or api?

### DISCUSSION THREAD

**Discussion by wadid** [lobehub/lobehub#4415] [Development Question]
Is there an api or usage over cli possible? E.g. i have a tool that sends a text input to lobechat over api, cli then it gets processed and lobechat then sends output to next tool?
I dont always want to go in chat interface but want to automate things. Still i would like to be able to access the ui view every now and then.

**Comment by arvinxx** [comment]
we have a plan to make the OpenAPI with lobechat server db. but the cli is not planned by now.

**Reply by wadid** [reply]
is there a feature difference between community and licensed edition?

**Reply by arvinxx** [reply]
we plan OpenAPI will only be offered in a paid edition. Common users who use Community Edtion don't need it , I think.

**Reply by GPTLocalhost** [reply]
> we plan OpenAPI will only be offered in a paid edition. Common users who use Community Edtion don't need it , I think.

It will be very helpful if API endpoints are available for Community Edtion and self-hosted Lobechat locally. I'm working on a local alternative to Microsoft's "Copilot in Word." It would be great to integrate Lobochat (as backend) with Word (as front end) through my local Word Add-in ([GPTLocalhost](https://gptlocalhost.com/demo/)). Demos for integrating Microsoft Word with AnywhereLLM (its workspaces) [here](https://gptlocalhost.com/demo/#anythingllm) and Ollama (its models) [here](https://gptlocalhost.com/demo/#ollama) are provided for your reference. 

Having Lobechat's APIs compatible with OpenAI's chat completion will be a good starting point. After that, a deeper integration with Microsoft Word can be made if Lobechat can provide more APIs, e.g. an API to generate an image based on input text. With the API and my Word Add-in, it will possible to pass the image back to Word based on some image caption in Word document. 

In brief, the point raised by @wadid can open a door for broader possibilities. Thank you for your consideration.

**Reply by arvinxx** [reply]
@GPTLocalhost LobeChat's community edition is aimed for personal usage, it's an alternative for ChatGPT / Claude.ai / Poe. The OpenAPI feature's more of developer/enterprise demand, so it's not included in the Community Edtion.

**Reply by wadid** [reply]
> It will be very helpful if API endpoints are available for Community Edtion and self-hosted Lobechat locally. I'm working on a local alternative to Microsoft's "Copilot in Word." It would be great to integrate Lobochat (as backend) with Word (as front end) through my local Word Add-in ([GPTLocalhost](https://gptlocalhost.com/demo/)). Demos for integrating Microsoft Word with AnywhereLLM (its workspaces) [here](https://gptlocalhost.com/demo/#anythingllm) and Ollama (its models) [here](https://gptlocalhost.com/demo/#ollama) are provided for your reference.

@GPTLocalhost AnywhereLLM might be what I am looking for. thx very much!

**Reply by GPTLocalhost** [reply]
> aimed for personal usage

I appreciate your reply. This is exactly what I'm aiming for as well. The target audience of my local Word Add-in is those who could not afford the price tag of Copilot in Word or prefer local inference and privacy or both. When the OpenAI-compatible endpoints are ready in Lobechat, I will be glad to integrate Lobechat (either Paid or Community) with Microsoft Word. Lobechat will be a powerful backend for Word users. Thank you for your consideration.

**Reply by arvinxx** [reply]
If you just need an  OpenAI-compatible endpoints, I really recommand you [litellm](https://docs.litellm.ai/docs/). LobeChat is not aimed at backend usage.

>  Lobechat will be a powerful backend for Word users.

in your usage, you are making lobechat as a service provider, it's not the proposed usage. What I mean personal usage is that user just use LobeChat interface.

**Reply by GPTLocalhost** [reply]
> LobeChat is not aimed at backend usage.
> What I mean personal usage is that user just use LobeChat interface.

Your clarification helps and thanks. My thinking is to let users use both LobeChat interface and Word interface together. My local Word Add-in is simply a bridge to integrate both worlds. If there are more API endpoints open for integration on LobeChat's end in the future, I'd be glad to know and try.
